openwebstart/src/main/java/com/openwebstart/proxy/pac/PacUtils.java

```diff
@@ -28,27 +30,28 @@
      * @return a list of Proxy objects representing the parsed string. In
      * case of malformed input, an empty list may be returned
      */
-    static List<Proxy> getProxiesFromPacResult(String pacString) {
-        final List<Proxy> proxies = new ArrayList<>();
-
-        final String[] tokens = pacString.split(";");
-        for (String token : tokens) {
-            if (token.startsWith(PROXY)) {
-                getInetSocketAddress(token)
-                        .map(sa -> new Proxy(Proxy.Type.HTTP, sa))
-                        .ifPresent(proxies::add);
-            } else if (token.startsWith(SOCKS)) {
-                getInetSocketAddress(token)
-                        .map(sa -> new Proxy(Proxy.Type.SOCKS, sa))
-                        .ifPresent(proxies::add);
-            } else if (token.startsWith(DIRECT)) {
-                proxies.add(Proxy.NO_PROXY);
-            } else {
-                LOG.debug("Unrecognized proxy token: {}", token);
-            }
+    static List<Proxy> getProxiesFromPacResult(final String pacString) {
+        if (pacString == null) {
+            return Collections.singletonList(Proxy.NO_PROXY);
         }
-
-        return proxies;
+        final String[] tokens = pacString.split(";");
+        return Stream.of(tokens)
+                .map(token -> token.trim())
+                .map(token -> {
+                    if (token.startsWith(PROXY)) {
+                        return getInetSocketAddress(token)
+                                .map(sa -> new Proxy(Proxy.Type.HTTP, sa))
+                                .orElseThrow(() -> new IllegalArgumentException("HTTP Proxy must be specified with valid address"));
+                    } else if (token.startsWith(SOCKS)) {
+                        return getInetSocketAddress(token)
+                                .map(sa -> new Proxy(Proxy.Type.SOCKS, sa))
+                                .orElseThrow(() -> new IllegalArgumentException("SOCKS Proxy must be specified with valid address"));
+                    } else if (token.startsWith(DIRECT)) {
+                        return Proxy.NO_PROXY;
+                    } else {
+                        throw new IllegalArgumentException("Unrecognized proxy token: " + token);
```

Comment:
This used to be a LOG.debug.
Now it throws and thus ending the processing of the stream
Could you LOG and return null and then in a later stage filter out any null values???

Reply:
I would say that an error by parsing the string should end in an error. If a pac file returns a string that does not match the definition of PAC results an exception should be thrown. Why return null in such case?